openmetadata-service/src/main/java/org/openmetadata/csv/CsvUtil.java

```diff
@@ -94,6 +102,65 @@ public static List<String> fieldToEntities(String field) {
     return field == null ? null : listOf(field.split(ENTITY_TYPE_SEPARATOR));
   }
 
+  public static List<String> fieldToInternalArray(String field) {
+    // Split a fieldValue that contains multiple elements of an array separated by
+    // INTERNAL_ARRAY_SEPARATOR
+    if (field == null || field.isBlank()) {
+      return Collections.emptyList();
+    }
+    return listOf(field.split(Pattern.quote(INTERNAL_ARRAY_SEPARATOR)));
+  }
+
+  /**
+   * Parses a field containing key-value pairs separated by semicolons, correctly handling quotes.
+   * Each key-value pair may also be enclosed in quotes, especially if it contains delimiter like (SEPARATOR , FIELD_SEPARATOR).
+   * Input Example:
+   * "key1:value1;key2:value2;\"key3:value;with;semicolon\""
+   * Output: [key1:value1, key2:value2, key3:value;with;semicolon]
+   *
+   */
+  public static List<String> fieldToExtensionStrings(String field) throws IOException {
+    if (field == null || field.isBlank()) {
+      return List.of();
+    }
+
+    // Step 1: Replace semicolons within quoted strings with a placeholder
```

Comment:
lets remove these comments, why do we need these @sonika-shah

Reply:
hi @harshach , the reason for adding comments on each line of the csv parser thing was so that its helpful for someone else in future to understand what are we doing and why we did the change.


if required , I will remove the changes